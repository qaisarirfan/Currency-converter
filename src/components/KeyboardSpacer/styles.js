import {StyleSheet} from "react-native"

// KeyboardSpacer: Styling here
const styles = StyleSheet.create({
  container: {
    left: 0,
    right: 0,
    bottom: 0,
  },
})

export default styles
