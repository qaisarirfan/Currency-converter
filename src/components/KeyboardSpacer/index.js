import React, {useState, useEffect} from "react"
import {Keyboard, View, Dimensions, Platform} from "react-native"
import PropTypes from "prop-types"
import styles from "./styles"

// KeyboardSpacer Component content
export const KeyboardSpacer = ({style, onToggle}) => {
  const [keyboardSpace, setKeyboardSpace] = useState(0)

  useEffect(() => {
    const updateKeyboardSpace = (event) => {
      if (!event.endCoordinates) {
        return
      }

      const screenHeight = Dimensions.get("window").height
      const newKeyboardSpace = screenHeight - event.endCoordinates.screenY
      setKeyboardSpace(newKeyboardSpace)
      onToggle(true, newKeyboardSpace)
    }
    const showEvt =
      Platform.OS === "android" ? "keyboardDidShow" : "keyboardWillShow"
    const showListener = Keyboard.addListener(showEvt, updateKeyboardSpace)

    const resetKeyboardSpace = () => {
      setKeyboardSpace(0)
      onToggle(false, 0)
    }
    const hideEvt =
      Platform.OS === "android" ? "keyboardDidHide" : "keyboardWillHide"
    const hideListener = Keyboard.addListener(hideEvt, resetKeyboardSpace)

    return () => {
      showListener.remove()
      hideListener.remove()
    }
  }, [])

  return <View style={[styles.container, {height: keyboardSpace}, style]} />
}

// KeyboardSpacer Proptypes
KeyboardSpacer.propTypes = {
  style: PropTypes.object,
  onToggle: PropTypes.func,
}

// KeyboardSpacer Default props
KeyboardSpacer.defaultProps = {
  style: {},
  onToggle: () => null,
}

export default KeyboardSpacer
