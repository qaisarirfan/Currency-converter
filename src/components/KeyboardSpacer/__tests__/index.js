import React from "react"
import renderer from "react-test-renderer"
import KeyboardSpacer from ".."

const createProps = () => ({
  classes: {},
})

describe("<KeyboardSpacer />", () => {
  it("Does not explode", () => {
    const props = createProps()
    const component = renderer.create(<KeyboardSpacer {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
