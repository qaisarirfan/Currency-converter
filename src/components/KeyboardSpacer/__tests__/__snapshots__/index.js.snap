// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyboardSpacer /> Does not explode 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
      },
      Object {
        "height": 0,
      },
      Object {},
    ]
  }
/>
`;
